ES6 Rewrite

mdlawson opened the pull request:
V1 is a full rewrite in ES6, with a new mechanism for including piping in your code, hooks for reloading events on both the supervisor and application process, and a test suite.

mdlawson committed:
Full rewrite in ES6, many big changes, see README.

mdlawson committed:
1.0.0-rc.1

mdlawson committed:
Support passing env to spawned process

mdlawson committed:
fix race condition which would allow multiple children to be spawned

mdlawson committed:
1.0.0-rc.3

mdlawson committed:
fixed event emitter leak with simultaneous file changes

mdlawson committed:
1.0.0-rc.4

sibelius commented:
any changes this will get in?

sibelius commented:
u should consider moving to Jest

https://github.com/skovhus/jest-codemods

it is much faster than ava

mdlawson commented:
Currently there's some race condition/leak in the newer version affecting node 4.4 (see #21) which I'd kinda like to get to the bottom of before merging this, in case it's an indication of some bigger issue. Unfortunately I haven't had a chance to look into it properly yet...

sibelius commented:
Tks for the answer. I think it is really hard to keep compatibility against different versions of node.

You should setup tests against many versions of node, like this repo here: https://github.com/chentsulin/koa-graphql/blob/master/.travis.yml

shahyar committed:
Fix ignoring resolved paths

Due to checking for only if a path starts with `"."`, or if it was in
`node_modules`, this code never actually hot reloaded if the file was in
a resolved path.

mdlawson committed:
Merge pull request #25 from shahyar/v1

Fix ignoring resolved paths

bertho-zero commented:
News of the v1?

bertho-zero commented:
Is there any news of the v1?
